src/Microsoft.Extensions.Logging.AzureWebApps/Internal/FileLoggerProvider.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Serilog;
+using Serilog.Formatting.Display;
+using Serilog.Sinks.RollingFile;
+
+namespace Microsoft.Extensions.Logging.AzureWebApps.Internal
+{
+    /// <summary>
+    /// A file logger for Azure WebApps
+    /// </summary>
+    public class FileLoggerProvider : WebAppsSerilogLoggerProvider
+    {
+        private const int RetainedFileCountLimit = 31; // Days
+        private const long FileSizeLimitBytes = 1L * 1024 * 1024 * 1024;
+        private const string OutputTemplate = "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level}] {Message}{NewLine}{Exception}";
+        private const string FileNamePattern = "log-{Date}.txt";
+
+
+        /// <summary>
+        /// Creates a new instance of the <see cref="FileLoggerProvider"/> class
+        /// </summary>
+        /// <param name="configReader">A configuration reader</param>
+        public FileLoggerProvider(IWebAppsLogConfigurationReader configReader)
+            : base(configReader)
+        {
+        }
+
+        /// <inheritdoc />
+        protected override void ConfigureLogger(LoggerConfiguration loggerConfiguration, WebAppsLogConfiguration webAppsConfiguration)
+        {
+            if (string.IsNullOrEmpty(webAppsConfiguration.FileLoggingFolder))
+            {
+                throw new ArgumentNullException(nameof(webAppsConfiguration.FileLoggingFolder), "The file logger path cannot be null");
+            }
+
+            var logsFolder = webAppsConfiguration.FileLoggingFolder;
+            if (!Directory.Exists(logsFolder))
+            {
+                Directory.CreateDirectory(logsFolder);
```

Comment:
Yes but this code is "public internal" and should not be called directly by the outside. Out "public public" API don't call it